contrib/hyperscan/matching/input_matchers/test/matcher_test.cc

```diff
@@ -46,6 +48,125 @@ TEST(MatcherTest, RaceScratchCreation) {
   }
 }
 
+// Verify that comparing works correctly for bounds.
+TEST(BoundTest, Compare) {
+  EXPECT_LT(Bound(1, 1), Bound(2, 1));
+  EXPECT_LT(Bound(1, 2), Bound(2, 1));
+  EXPECT_LT(Bound(2, 2), Bound(2, 1));
+}
+
+class MatcherTest : public ::testing::Test {
+protected:
+  void setup(const char* expression, unsigned int flag, bool report_start_of_matching) {
```

Comment:
Can you expose a helper function that translates protobufs into the flag bitmask and test using the protobufs?

Reply:
In fact, no. The matcher test works for both Hyperscan input matcher and regex engine, so introduce proto here is not reasonable. If you think it is necessary to test protos, we should add them in config_test.cc

All above replied or fixed, thanks.